source/packages/services/assetlibrary-export/src/batch/batch.service.ts

```diff
@@ -38,16 +41,22 @@ export class BatchService implements Batcher {
 
     public async batch(): Promise<Batches> {
         const batches = await this.batchers[this.batchBy].batch();
+        const s3Saves:Promise<AWS.S3.PutObjectOutput>[] = [];
+
+        const limit = pLimit(this.PROMISE_CONCURRENCY_LIMIT);
 
         // save individual batches to S3 for etl
         // optimization for step function
-        for (const batch of batches) {
+        for (const index in batches) {
+            const batch = batches[index];
+            batch.id = index;
             const key = `${this.exportKeyPrefix}_temp/${batch.id}`
-            await this.s3Utils.save(this.exportBucket, key, batch);
+            s3Saves.push(limit(() => this.s3Utils.save(this.exportBucket, key, batch)));
         }
+        // concurrently save the individual batches to s3
+        await Promise.all(s3Saves);
```

Comment:
With `Promise.all`, is one fails then the whole thing fails. Is that the intended behavior? Or should it be using `Promise.allSettled`?